### ISSUE TITLE
[JAX backend + possibly others] `generate()` decode error

### ISSUE THREAD

**Issue by abheesht17** [keras-team/keras-nlp#1149]
Do not see the same error in legacy Keras. Secondly, there is a huge difference in loss (JAX at 5.5), whereas Keras is at 3.something. So, I guess the model is not getting trained as well as the legacy Keras one, and is producing gibberish. I do not see this error when I train on more data. But in any case, we really need to sort out this `decode` issue

Repro: https://colab.research.google.com/drive/1TyEUS9fCJBq9duZMgOXJlchYsvH2r5Vx?usp=sharing

**Comment by mattdangerw** [comment]
I think we fixed this with a combo of https://github.com/keras-team/keras-nlp/pull/1150 and landing a loss scale optimizer in Keras Core/Keras 3. Let's reopen if I am mistaken!
